src/org/thoughtcrime/securesms/mms/MediaConstraints.java

```diff
@@ -83,4 +98,39 @@ public MediaStream getResizedMedia(@NonNull Context context,
       throw new IOException(e);
     }
   }
+
+ public MediaStream compressFile(@NonNull Context context, @NonNull MasterSecret masterSecret, @NonNull Attachment attachment) throws IOException, URISyntaxException {
+   if (!MediaUtil.isVideo(attachment) && !MediaUtil.isImage(attachment) && !MediaUtil.isGif(attachment)) {
+           throw new UnsupportedOperationException("File type not video or image or gif! Cannot compress");
+   }
+
+   String directory = context.getCacheDir().toString() + "/tempFile" + attachment.getContentType().replace('/', '.');
+   InputStream is = PartAuthority.getAttachmentStream(context, masterSecret, attachment.getDataUri());
+   byte[] byteStream = Util.readFully(is);
+
+   FileOutputStream outputStream = new FileOutputStream(directory);
+   outputStream.write(byteStream);
+
+   String filepath = "";
+
+   if(MediaUtil.isVideo(attachment)) {
+     filepath = compressVideo(context, directory);
+   } else {
+     filepath = compressImage(context, directory);
+   }
+
+   outputStream.close();
+   is.close();
+   new File(directory).delete();
+
+   return new MediaStream(new FileInputStream(filepath), attachment.getContentType());
+ }
+
+ public String compressVideo(@NonNull Context context, String directory) throws URISyntaxException, FileNotFoundException {
```

Comment:
`compressVideo` should be private if it is only called from within the class

Reply:
Being used in compressionTest.java